Launch the domain step test

niranjan-uma-shankar opened the pull request:
#### Changes proposed in this Pull Request

* Launch the domain step copy update test `domainStepCopyUpdates` to 50% users and remove feature flag( introduced in #37546)
* The copy updates will only be seen in the signup flow, and not in launch flow or in the "Add domain" page.

**Web**

<img width="989" alt="Screenshot 2019-11-21 at 12 25 26 PM" src="https://user-images.githubusercontent.com/1269602/69314419-1e1c9100-0c5a-11ea-87d4-fa3f24c20b5b.png">

**Mobile**

![calypso localhost_3000_start_domains-with-preview(iPhone 5_SE)](https://user-images.githubusercontent.com/1269602/69314443-2d9bda00-0c5a-11ea-9a03-43a54e9db844.png)


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go through the signup flow at /start
* If you are assigned to the `control` group of the `domainStepCopyUpdates` test, then verify the following:
    - The domain step UI and copy remains the same as before
    - Complete signup and check the launch flow steps - they should remain unchanged
* If you are assigned to the `variantShowUpdates` group, then verify the following:
    - The UI and copy for the domain step should match the screenshots above
    - Selecting the "Review plan options" cta should show the plan page with _free site option hidden_.
    - Selecting the Free *.wordpress.com(or *.home.blog etc) domain should show the free site option in the plan page. Selecting a custom domain should hide the free site option. This is same as the current behaviour.
    - The new UI and copy _should not show_ in the site launch flow or in the Add domain page.
    - Add new site should show the new UI and copy. 

**A/B test assignment**

1. If you drop off at any step before the domain step, verify that you do not get assigned to the A/B test. Verify this by typing `localStorage.ABTests` in your browser console and checking that the `domainStepCopyUpdates` is not present in the object.

2. Sign in to an existing account and go through the launch flow or view the Add domain page. Neither of these should assign you to the A/B test.

dzver commented:
Code LGTM

The test variation is a bit too motley for my taste. Hiding the Free plan also doesn't make too much sense to me. We've proved on multiple occasions that hiding the free plan is not good because people don't buy more just because we obstruct them from signing up, they just don't sign up.

niranjan-uma-shankar commented:
Thanks for the review @dzver . I'll share your feedback with the team. If you'd like to share further feedback, feel free to ping the folks in the slack channel #domain-step-optimization.

obi2020 commented:
The text seems longer than it a needs to be. A suggestion:

### Your domain is free for a year with any paid plan

Not ready to choose? Buy your plan now and you can choose your domain later. [Compare all plans >>](url)

niranjan-uma-shankar commented:
Made a couple of minor fixes:

1. In [c0a32a3](https://github.com/Automattic/wp-calypso/pull/37661/commits/c0a32a3bd1a0771868a7e8f9a95f9b795fc9405a), I fixed the PropTypes for the `Badge` component, added a better name for the badge type name, and added a unit test. Without the PropTypes fix, a Warning message was displayed in console:
```
backend.js:6 Warning: Failed prop type: Invalid prop `type` of value `blue` supplied to `Badge`, expected one of ["warning","success","info"].
```
2. In Included the `shouldHideFreePlan ` optional dependency so that the /start/with-theme flow works as expected.

niranjan-uma-shankar commented:
@obi2020 Thanks for the suggestion Obi. I've posted your feedback in #domain-step-optimization in slack.
